language/diem-framework/modules/DiemId.move

```diff
@@ -83,5 +85,25 @@ module DiemId {
         exists<DiemIdDomains>(addr)
     }
 
+    public fun update_diem_id_domain(tc_account: &signer, to_update_address: address, domain: vec<u8>, is_remove: bool) {
```

Comment:
I can't comment on the line I want to but is it possible to enforce validation at the move level? and do something similar to this:

```
pub struct VaspDomainIdentifier(Box<str>);
pub struct VaspDomainIdentifierError;
impl VaspDomainIdentifier {
    pub fn new(s: &str) -> Result<Self, VaspDomainIdentifierError> {
        if s.len() > 63 || s.is_empty() {
            return Err(VaspDomainIdentifierError);
        }
        let mut chars = s.chars();
        match chars.next() {
            Some('a'..='z') | Some('A'..='Z') | Some('0'..='9') => {}
            Some(_) => return Err(VaspDomainIdentifierError),
            None => return Err(VaspDomainIdentifierError),
        }
        for c in chars {
            match c {
                'a'..='z' | 'A'..='Z' | '0'..='9' | '.' => {}
                _ => return Err(VaspDomainIdentifierError),
            }
        }
        Ok(VaspDomainIdentifier(s.into()))
    }
}
```

Reply:
Might add support for it in a follow-up PR, not sure if enforcing this in move makes sense atm